xarray/backends/rasterio_.py

```diff
@@ -313,6 +313,11 @@ def open_rasterio(filename, parse_coordinates=None, chunks=None, cache=None,
     if hasattr(riods, 'units') and any(riods.units):
         # A list of units string for each dataset band
         attrs['units'] = riods.units
+    if hasattr(riods, 'mask_flag_enums'):
+        # Sets of flags describing the sources of band masks
+        attrs['mask_flags'] = [
```

Comment:
Let's see what @fmaussion thinks about the string approach. I don't use rasterio so I don't know what that community would find useful.

Reply:
I don't use these attributes either, unfortunately. The string trick doesn't look very good though...

What is the expected dimensions of this attribute? Can it be anything, or is it bound to the dimension of the bands? Because this looks like one of the limitation of the DataArray model for rasterio objects (another possibility would have been a Dataset, it was discussed in earlier stages but then discarded).

Reply:
> The string trick doesn't look very good though...
:+1: 

I prefer to use the int bit mask as defined in GDAL documentation (rasterio expose these ints too), but it can be a little puzzling for people who dont known gdal internals.

Reply:
Thanks. Let's just not deal with `mask_flag_enums` unless someone really needs it. None of these approaches seem that great unfortunately.

Reply:
Instead of incrementally adding the attributes, could there be a way to expose the rasterio object *from* the DataArray object? I guess there is no clean way to do so, but it would allow everyone to access whatever they want...

Reply:
> Thanks. Let's just not deal with `mask_flag_enums` unless someone really needs it. None of these approaches seem that great unfortunately.

@dcherian I agree with you. Today `mask_flag_enums` is useless for me, and probably someone that really uses it can find a better way to expose the attribute.